version: "3.8"

services:
  app:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: c-producto-app-prueba
    ports:
      - "5000:3000"
    env_file:
      - .env
    volumes:
      - .:/usr/src/app
      - /usr/src/app/node_modules
    depends_on:
      mysql:
        condition: service_healthy
    networks:
      - mysql_net

  mysql:
    image: mysql:8.0
    container_name: c-mysql_container
    environment:
      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
      MYSQL_DATABASE: ${MYSQL_DATABASE}
      MYSQL_USER: ${MYSQL_USER}
      MYSQL_PASSWORD: ${MYSQL_PASSWORD}
    ports:
      - "5001:3306"
    volumes:
      - mysql_dt:/var/lib/mysql
    networks:
      - mysql_net
    healthcheck:
      test: ["CMD-SHELL", "mysqladmin ping -h 127.0.0.1 --silent"]
      interval: 10s
      timeout: 5s
      retries: 3

volumes:
  mysql_dt:

networks:
  mysql_net:
